packages/api/src/indexPriceInterface.ts

```diff
@@ -120,20 +124,31 @@ export default abstract class IndexPriceInterface extends Observer {
 		}
 	}
 
-	private async _onRedisFeedHandlerMsg(message: string) {
+	public async _onRedisFeedHandlerMsg(message: string) {
 		// message must be indices separated by semicolon
 		// console.log("Received REDIS message" + message);
 		const indices = message.split(";");
+
+		// Create new updated indices and only send those that were updated.
+		const updatedIndices: string[] = [];
+
 		for (let k = 0; k < indices.length; k++) {
 			// get price from redit
-			const px_ts = await this.redisClient.ts.get(indices[k]);
-			const px = px_ts?.value;
-			if (px != undefined) {
-				this.idxPrices.set(indices[k], px);
+			try {
+				const px_ts = await this.redisClient.ts.get(indices[k]);
+				const px = px_ts?.value;
+				if (px != undefined) {
+					this.idxPrices.set(indices[k], px);
+				}
+				updatedIndices.push(indices[k]);
```

Comment:
do we need to add indices[k] to updatedIndices if the price px is undefined?

Reply:
`px` can only be undefined if `ts.get` call fails, which would trigger try/catch with `ERR TSDB: the key does not exist`

Reply:
ok so then the lines 140 to 142 are useless and they should be removed

Reply:
`px_ts` can actually be `null` if there are no entries in the timeseries, but the key exists. Updated.